dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -315,9 +321,144 @@ public void saveScheme(final WorkflowScheme scheme, final User user) throws DotD
 
 	}
 
+
+	public void deleteScheme(final WorkflowScheme scheme, final User user)
+			throws DotDataException, DotSecurityException, AlreadyExistException {
+
+		if (null == scheme || SYSTEM_WORKFLOW_ID.equals(scheme.getId()) || !scheme.isArchived()) {

```

Comment:
I think this  `null == scheme ` validation opens the door for a NPE. 
That can arise from operations like `scheme.getId()` or `scheme.isArchived()`
if scheme is null the condition becomes true and we could get a NPE on line 330